v1/api/index.ts

```diff
@@ -0,0 +1,69 @@
+import { TypeBoxTypeProvider } from '@fastify/type-provider-typebox'
+import fastify, { FastifyBaseLogger, FastifyInstance, RawReplyDefaultExpression, RawRequestDefaultExpression, RawServerDefault } from 'fastify'
+import { siteRoutes } from './sites'
+import multipart from '@fastify/multipart'
+import swagger from '@fastify/swagger'
+import swagger_ui from '@fastify/swagger-ui'
+import { adminRoutes } from './admin'
+import { publisherRoutes } from './publisher'
+
+export type FastifyTypebox = FastifyInstance<
+RawServerDefault,
+RawRequestDefaultExpression<RawServerDefault>,
+RawReplyDefaultExpression<RawServerDefault>,
+FastifyBaseLogger,
+TypeBoxTypeProvider
+>
+
+interface APIConfig {
+  useLogging: boolean
+  useSwagger: boolean
+}
+
+async function apiBuilder ({ useLogging, useSwagger }: Partial<APIConfig>): Promise<FastifyTypebox> {
+  const server = fastify({ logger: useLogging }).withTypeProvider<TypeBoxTypeProvider>()
+  await server.register(multipart) // TODO: discuss whether we want to set a filesize limit
+
+  server.get('/healthz', () => {
+    return 'ok\n'
+  })
+
+  await server.register(v1Routes(useSwagger ?? false), { prefix: '/v1' })
+  await server.ready()
+  return server
+}
+
+const v1Routes = (useSwagger: boolean) => async (server: FastifyTypebox): Promise<void> => {
+  if (useSwagger) {
```

Comment:
TBH I don't think we need to put this behind an `if(useSwagger)` and just expose it for everyone. Is there a use case you had in mind where swagger should be disabled?

Reply:
this is mostly so we don't build Swagger for tests so they run faster :')) same reason why there is a flag for logging